fpdf/html.py

```diff
@@ -558,6 +563,10 @@ def handle_starttag(self, tag, attrs):
         LOGGER.debug("STARTTAG %s %s", tag, attrs)
         parse_style(attrs)
         self._tags_stack.append(tag)
+        if attrs.get("page-break-before") == "always":
```

Comment:
Are you sure this does what you think it does?

You're looking for a HTML attribute named "page-break-before", which doesn't exist (similar to the currently still incorrect use of "line-height" in this module).
The correct approach would be to find a "style" attribute, and scan its value for a "break-before: always;" substring.

You're also using an outdate name. According to the standard, the CSS property is named "break-before".

Reply:
You are right regarding the deprecated attribute name, I'm going to fix that.

However, as can prove the unit tests, this code is correctly checking either the CSS property or a similarly named HTMl attribute, thanks to the call to `parse_style()` above: https://github.com/py-pdf/fpdf2/blob/2.7.9/fpdf/html.py#L542

Reply:
Apparently I didn't pay enough attention when you added `parse_style()`.

However, I already pointed out [in this issue](https://github.com/py-pdf/fpdf2/issues/1116#issuecomment-1960109239_), that eg. "line-height" is not a valid HTML attribute, and we should not process it as if it was. Despite that, we currently do accept and process it.

If I understand the current implementation correctly, then HTML attributes and CSS properties are just thrown together in a bucket as if they were the same. The consequence is that any key-value pair that we accept as one, is automatically also accepted as the other. This is something we absolutely must NOT do. We want to process neither invalid HTML attributes nor invalid CSS properties, both of which seems to be the case at the moment.

If we want to accept CSS properties, then we need to do this differently. First, HTML attributes and CSS properties must be collected into seperate dicts. Then for each HTML tag, we can query both dicts, each for the respectively appropriate keys. I don't actually think that this will be very difficult to implement, but it will be an enormous quality improvement for the HTML module.

Or did I miss some hidden magic that already distinguishes between the two?

Reply:
> If I understand the current implementation correctly, then HTML attributes and CSS properties are just thrown together in a bucket as if they were the same.
> [...] Or did I miss some hidden magic that already distinguishes between the two?

Yes, this is exactly what happens since PR #1141 in April.

> We want to process neither invalid HTML attributes nor invalid CSS properties, both of which seems to be the case at the moment.
> If we want to accept CSS properties, then we need to do this differently. First, HTML attributes and CSS properties must be collected into seperate dicts. Then for each HTML tag, we can query both dicts, each for the respectively appropriate keys.

I agree.
I'm OK to take the point to create a distinct PR to perform this rafactoring, after this one is merged.

Reply:
PR open: https://github.com/py-pdf/fpdf2/pull/1211